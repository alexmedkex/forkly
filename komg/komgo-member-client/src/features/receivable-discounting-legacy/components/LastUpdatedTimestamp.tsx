import React from 'react'
import moment from 'moment-timezone'
import styled from 'styled-components'

interface ILastUpdatedTimestampProps {
  date: string
}

export const LastUpdatedTimestamp: React.FC<ILastUpdatedTimestampProps> = ({ date }) => (
  <SmallSpan>
    <b>Last updated: </b>
    {moment(date).format('DD/MM/YYYY [at] HH:mm z')}
  </SmallSpan>
)

const SmallSpan = styled.div`
  font-family: LotaGrotesque;
  font-size: 14px;
  padding: 10px 0px;
  b {
    font-size: 13px;
  }
`
