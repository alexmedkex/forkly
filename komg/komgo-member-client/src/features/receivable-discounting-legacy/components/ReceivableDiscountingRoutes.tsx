import * as React from 'react'
import { Route, Switch } from 'react-router-dom'

import ApplyForDiscountingContainer from '../../receivable-finance/pages/apply-for-discounting/ApplyForDiscountingContainer'
import PushToMarketContainer from '../../receivable-finance/pages/push-to-market/PushToMarketContainer'
import ReceivableFinanceDashboardContainer from '../../receivable-finance/pages/dashboard/ReceivableFinanceDashboardContainer'
import DiscountingRequestViewContainer from '../../receivable-finance/pages/view-request/DiscountingRequestViewContainer'
import ViewQuotesContainer from '../../receivable-finance/pages/view-quotes/ViewQuotesContainer'
import AcceptQuoteContainer from '../../receivable-finance/pages/accept-quote/AcceptQuoteContainer'
import SubmitQuoteContainer from '../../receivable-finance/pages/submit-quote/SubmitQuoteContainer'
import { withPadding } from '../../../routes'

export const ReceivableDiscountingRoutes = () => (
  <Switch>
    <Route
      exact={true}
      path="/receivable-discounting"
      render={() => withPadding(<ReceivableFinanceDashboardContainer />)}
    />
    <Route path="/receivable-discounting/:tradeId/apply" component={ApplyForDiscountingContainer} />
    <Route path="/receivable-discounting/:rdId/request-for-proposal" component={PushToMarketContainer} />
    <Route path="/receivable-discounting/:rdId/quotes" component={ViewQuotesContainer} />
    <Route exact={true} path="/receivable-discounting/:rdId" component={DiscountingRequestViewContainer} />
    <Route path="/receivable-discounting/:rdId/provide-quote" component={SubmitQuoteContainer} />
    <Route path="/receivable-discounting/:rdId/accept/:participantStaticId" component={AcceptQuoteContainer} />
  </Switch>
)

export default ReceivableDiscountingRoutes
