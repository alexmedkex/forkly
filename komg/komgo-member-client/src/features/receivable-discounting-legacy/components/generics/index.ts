export * from './FixedButtonBar'
export * from './Page'
