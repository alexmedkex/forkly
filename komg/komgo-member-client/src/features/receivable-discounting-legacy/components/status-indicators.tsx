import * as React from 'react'
import { Label } from 'semantic-ui-react'
import styled from 'styled-components'
import { sentenceCase } from '../../../utils/casings'
import {
  PARTICIPANT_RFP_STATUS_TO_COLOR,
  RD_STATUS_TO_COLOR
} from '../../receivable-finance/entities/rd/constants/status-to-color'
import { displayTradeStatus } from '../../trades/utils/displaySelectors'

interface IProps {
  status: string
}

export const RFPSummaryStatusIndicator: React.FC<IProps> = ({ status }) => {
  const color = (PARTICIPANT_RFP_STATUS_TO_COLOR as any)[status]

  const LabelWrapper = styled(Label)`
    &&& {
      background-color: ${color};
      border-color: ${color};
      color: white;
    }
  `
  return (
    <LabelWrapper as="span" data-test-id="rfp-summary-status-span">
      {sentenceCase(status)}
    </LabelWrapper>
  )
}

export const RDStatusIndicator: React.FC<IProps> = ({ status }) => {
  const color = (RD_STATUS_TO_COLOR as any)[status]

  const LabelWrapper = styled(Label)`
    &&& {
      background-color: ${color};
      border-color: ${color};
      color: white;
    }
  `
  return (
    <LabelWrapper as="span" data-test-id="rd-status-span">
      {sentenceCase(status)}
    </LabelWrapper>
  )
}

export const TopbarStatusIndicator: React.FC<IProps> = ({ status }) => {
  return (
    <>
      <p>STATUS: </p>
      <RDStatusIndicator status={displayTradeStatus(status)} />
    </>
  )
}
