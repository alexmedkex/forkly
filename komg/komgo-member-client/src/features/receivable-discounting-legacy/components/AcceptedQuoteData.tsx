import { IQuote, IQuoteBase, IHistory, RequestType, DiscountingType, IReceivablesDiscounting } from '@komgo/types'
import moment from 'moment'
import * as React from 'react'
import styled from 'styled-components'
import { MinimalAccordionWrapper } from '../../../components/accordion/MinimalAccordionWrapper'
import { DisplayQuote } from '../../receivable-finance/entities/quote/components/DisplayQuote'
import { Button } from 'semantic-ui-react'
import { Formik, FormikProps } from 'formik'
import { EditAcceptedQuoteFields } from '../../receivable-finance/pages/view-request/components/EditAcceptedQuoteFields'
import { MultiErrorMessage } from '../../../components/error-message'
import { formikQuoteAcceptedErrors } from '../utils/errors'
import { decorateQuoteForInitialValues, createEditQuoteValidator } from '../utils/edit-utils'
import { StatusText } from './generics/StatusText'

export interface IAcceptedQuoteData {
  index: string
  open: boolean
  quote: IQuote
  isSubmitting: boolean
  isEditing: boolean
  editable: boolean
  comment?: string
  replyDate?: string
  provider?: string
  agreedTermsHistory: IHistory<IQuote>
  changed: boolean
  rd: IReceivablesDiscounting
  handleSubmit: (values: IQuoteBase) => void
  handleEditClicked: () => void
  handleCancelClicked: () => void
  handleToggleAccordion: () => void
}

const AcceptedQuoteData: React.FC<IAcceptedQuoteData> = ({
  index,
  open,
  quote,
  comment,
  replyDate,
  provider,
  editable,
  isEditing,
  changed,
  isSubmitting,
  handleCancelClicked,
  handleEditClicked,
  handleToggleAccordion,
  handleSubmit,
  agreedTermsHistory,
  rd
}) => {
  const initialValues = quote ? decorateQuoteForInitialValues(quote) : {}
  const actions = formik =>
    isEditing ? (
      <>
        <Button
          content={'Save'}
          data-test-id="save-accepted-quote"
          primary={true}
          type="submit"
          disabled={isSubmitting}
          onClick={e => {
            e.stopPropagation()
            formik.handleSubmit(e)
          }}
        />
        <Button
          content={'Cancel'}
          data-test-id="savecancel-edit-accepted-quote"
          disabled={isSubmitting}
          onClick={e => {
            e.stopPropagation()
            formik.resetForm()
            handleCancelClicked()
          }}
        />
      </>
    ) : editable ? (
      <Button
        content={'Edit'}
        data-test-id="edit-accepted-quote"
        onClick={e => {
          e.stopPropagation()
          handleEditClicked()
        }}
      />
    ) : null

  return (
    <Formik
      initialValues={initialValues}
      onSubmit={handleSubmit}
      validate={createEditQuoteValidator(initialValues, rd)}
      validateOnBlur={false}
      validateOnChange={true}
      render={(formik: FormikProps<IQuoteBase>) => (
        <>
          <MinimalAccordionWrapper
            buttons={actions(formik)}
            active={open}
            handleClick={handleToggleAccordion}
            index={index}
            highlight={changed}
            title="Agreed terms"
          >
            {formikQuoteAcceptedErrors(formik).length > 0 ? (
              <MultiErrorMessage
                data-test-id="edit-quote-accepted-validation-errors"
                title=""
                messages={formikQuoteAcceptedErrors(formik)}
              />
            ) : null}
            <StatusText data-test-id="quote-section-status-text" margin="0 0 0 28px">
              <UpperCaseText>Quote accepted - {moment(quote.createdAt).format('DD/MM/YYYY [at] HH:mm')}</UpperCaseText>
            </StatusText>
            {isEditing ? (
              <EditAcceptedQuoteFields data-test-id="edit-accepted-quote-panel" formik={formik} rd={rd} />
            ) : (
              <DisplayQuote
                quote={quote}
                provider={provider}
                comment={comment}
                replyDate={replyDate}
                agreedTermsHistory={agreedTermsHistory}
                sectionName="agreedTerms"
                requestType={rd.requestType}
                discountingType={rd.discountingType}
              />
            )}
          </MinimalAccordionWrapper>
        </>
      )}
    />
  )
}

const UpperCaseText = styled.span`
  &&& {
    text-transform: uppercase;
  }
`

export default AcceptedQuoteData
