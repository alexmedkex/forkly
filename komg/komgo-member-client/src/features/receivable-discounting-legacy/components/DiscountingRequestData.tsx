import {
  IReceivablesDiscountingBase,
  IReceivablesDiscountingInfo,
  IHistory,
  IReceivablesDiscounting,
  RequestType
} from '@komgo/types'
import { Formik, FormikProps } from 'formik'
import React from 'react'
import { Button } from 'semantic-ui-react'
import { MinimalAccordionWrapper } from '../../../components/accordion/MinimalAccordionWrapper'
import { MultiErrorMessage } from '../../../components/error-message'
import BasicPanel from '../../trades/components/BasicPanel'
import { formikRdErrors } from '../utils/errors'
import { EditDiscountingRequestFields } from '../../receivable-finance/pages/view-request/components/EditDiscountingRequestFields'
import { DiscountingRequestInfo } from './panels/DiscountingRequestInfo'
import { decorateRDForInitialValues, createReceivableDiscountingEditValidator } from '../utils/edit-utils'
import { Dimensions } from '../resources/dimensions'
import { displaySimpleRequestType } from '../utils/displaySelectors'
import { FieldDataProvider, FieldDataContext } from '../presentation/FieldDataProvider'
import { rdDiscountingSchema, initialApplyForDiscountingData } from '../utils/constants'

export interface IDiscountingRequestDataProps {
  discountingRequest: IReceivablesDiscountingInfo
  index: string
  open: boolean
  isSubmitting: boolean
  isEditing: boolean
  editable: boolean
  changed: boolean
  isLoadingHistory: boolean
  history: IHistory<IReceivablesDiscounting>
  handleSubmit: (values: IReceivablesDiscountingBase) => void
  handleEditClicked: () => void
  handleCancelClicked: () => void
  handleToggleAccordion: () => void
}

export const DiscountingRequestData: React.FC<IDiscountingRequestDataProps> = ({
  index,
  discountingRequest,
  open,
  editable,
  isEditing,
  isSubmitting,
  changed,
  handleCancelClicked,
  handleEditClicked,
  handleSubmit,
  handleToggleAccordion,
  history
}) => {
  const initialValues = decorateRDForInitialValues(discountingRequest.rd)
  const actions = formik =>
    isEditing ? (
      <>
        <Button
          content={'Save'}
          data-test-id="save-discounting-request"
          primary={true}
          type="submit"
          disabled={isSubmitting}
          onClick={e => {
            e.stopPropagation()
            formik.handleSubmit(e)
          }}
        />
        <Button
          content={'Cancel'}
          data-test-id="cancel-edit-discounting-request"
          disabled={isSubmitting}
          onClick={e => {
            e.stopPropagation()
            formik.resetForm()
            handleCancelClicked()
          }}
        />
      </>
    ) : editable ? (
      <Button
        content={'Edit'}
        data-test-id="edit-discounting-request"
        onClick={e => {
          e.stopPropagation()
          handleEditClicked()
        }}
      />
    ) : null

  const title = `${displaySimpleRequestType(discountingRequest.rd.requestType)} data`

  const fieldDataProvider = new FieldDataProvider(
    rdDiscountingSchema,
    initialApplyForDiscountingData(discountingRequest.rd.requestType, discountingRequest.rd.discountingType)
  )

  return (
    <Formik
      initialValues={initialValues}
      onSubmit={handleSubmit}
      validate={createReceivableDiscountingEditValidator(initialValues)}
      validateOnBlur={false}
      validateOnChange={true}
      render={(formik: FormikProps<IReceivablesDiscountingBase>) => (
        <>
          <MinimalAccordionWrapper
            data-test-id="discounting-request-accordion"
            active={open}
            handleClick={handleToggleAccordion}
            index={index}
            title={title}
            buttons={actions(formik)}
            path={'/'}
            highlight={changed || false}
          >
            {formikRdErrors(formik).length > 0 ? (
              <MultiErrorMessage
                data-test-id="edit-request-validation-errors"
                title=""
                messages={formikRdErrors(formik)}
              />
            ) : null}
            <BasicPanel padding={Dimensions.ViewRequestSectionBasicPanelPadding}>
              {isEditing ? (
                <FieldDataContext.Provider value={fieldDataProvider}>
                  <EditDiscountingRequestFields data-test-id="edit-discounting-request-panel" formik={formik} />
                </FieldDataContext.Provider>
              ) : (
                <DiscountingRequestInfo
                  data-test-id="view-discounting-request-panel"
                  history={history}
                  discountingRequest={discountingRequest}
                />
              )}
            </BasicPanel>
          </MinimalAccordionWrapper>
        </>
      )}
    />
  )
}

export default DiscountingRequestData
