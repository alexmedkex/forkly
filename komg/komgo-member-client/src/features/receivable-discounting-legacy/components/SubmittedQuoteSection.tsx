import { IQuote, IReceivablesDiscountingInfo } from '@komgo/types'
import moment from 'moment'
import * as React from 'react'

import { MinimalAccordionWrapper } from '../../../components/accordion/MinimalAccordionWrapper'
import { DisplayQuote } from '../../receivable-finance/entities/quote/components/DisplayQuote'
import { UpperCaseText } from './generics/UpperCaseText'
import { StatusText } from './generics/StatusText'

export interface ISubmittedQuoteSectionProps {
  discountingRequest: IReceivablesDiscountingInfo
  index: string
  isSubmittedQuoteAccordionOpen: boolean
  submittedQuote: IQuote
  comment?: string
  handleClick: (e: React.SyntheticEvent, titleProps: any) => void
}

export default class SubmittedQuoteSection extends React.Component<ISubmittedQuoteSectionProps> {
  render() {
    const {
      index,
      handleClick,
      isSubmittedQuoteAccordionOpen,
      submittedQuote,
      comment,
      discountingRequest
    } = this.props

    const submittedQuoteTimestamp = (quote: IQuote): string => moment(quote.createdAt).format('DD/MM/YYYY [at] HH:mm')

    return (
      <MinimalAccordionWrapper
        active={isSubmittedQuoteAccordionOpen}
        handleClick={handleClick}
        index={index}
        title="Submitted quote"
      >
        <StatusText data-test-id="submitted-quote-section-status-text" margin="0 0 0 28px">
          <UpperCaseText>Quote submitted - {submittedQuoteTimestamp(submittedQuote)}</UpperCaseText>
        </StatusText>
        <DisplayQuote
          quote={submittedQuote}
          comment={comment}
          sectionName="submittedQuote"
          requestType={discountingRequest.rd.requestType}
          discountingType={discountingRequest.rd.discountingType}
        />
      </MinimalAccordionWrapper>
    )
  }
}
