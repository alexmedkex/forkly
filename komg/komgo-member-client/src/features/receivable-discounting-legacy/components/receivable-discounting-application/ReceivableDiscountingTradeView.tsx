import * as React from 'react'
import { MinimalAccordionWrapper } from '../../../../components/accordion/MinimalAccordionWrapper'
import { PANELS, TradeViewData } from '../../../trades/components'
import { ITradeEnriched } from '../../../trades/store/types'
import { ICargo, IHistory, ITradeSnapshot } from '@komgo/types'
import { Button, AccordionTitleProps } from 'semantic-ui-react'

interface IReceivableDiscountingTradeViewProps {
  company: string
  trade: ITradeEnriched
  tradeMovements: ICargo[]
  open: boolean
  index: string
  changed: boolean
  tradeCargoHistory?: IHistory<ITradeSnapshot>
  handleClick: (e: React.SyntheticEvent, titleProps: AccordionTitleProps) => void
  handleEditClicked?: () => void
}

interface IReceivableDiscountingTradeViewState {
  tradeViewPanelActives: { [key in PANELS]: boolean }
}

export class ReceivableDiscountingTradeView extends React.Component<
  IReceivableDiscountingTradeViewProps,
  IReceivableDiscountingTradeViewState
> {
  constructor(props: IReceivableDiscountingTradeViewProps) {
    super(props)

    this.state = {
      tradeViewPanelActives: {
        [PANELS.Basic]: true,
        [PANELS.Goods]: true,
        [PANELS.Contract]: true,
        [PANELS.Terms]: true,
        [PANELS.Documents]: false,
        [PANELS.UploadedDocuments]: false,
        [PANELS.Cargo]: true
      }
    }
  }

  render() {
    const {
      company,
      trade,
      tradeMovements,
      open,
      index,
      changed,
      tradeCargoHistory,
      handleClick,
      handleEditClicked
    } = this.props
    const { tradeViewPanelActives } = this.state
    return (
      <MinimalAccordionWrapper
        active={open}
        handleClick={handleClick}
        index={index}
        title="Trade summary"
        highlight={changed}
        buttons={
          handleEditClicked && (
            <Button
              content={'Edit'}
              data-test-id="edit-trade-request"
              onClick={e => {
                e.stopPropagation()
                handleEditClicked()
              }}
            />
          )
        }
      >
        <TradeViewData
          company={company}
          actives={tradeViewPanelActives}
          hideDropdownIcon={true}
          trade={trade}
          tradeMovements={tradeMovements}
          tradeCargoHistory={tradeCargoHistory}
        />
      </MinimalAccordionWrapper>
    )
  }
}

export default ReceivableDiscountingTradeView
