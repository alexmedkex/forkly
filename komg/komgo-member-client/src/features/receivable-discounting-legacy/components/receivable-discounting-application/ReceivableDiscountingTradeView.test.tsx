import { shallow, mount, ReactWrapper, ShallowWrapper } from 'enzyme'
import * as React from 'react'
import { MinimalAccordionWrapper } from '../../../../components/accordion/MinimalAccordionWrapper'
import { TradeViewData } from '../../../trades/components'
import ReceivableDiscountingTradeView from './ReceivableDiscountingTradeView'
import { BrowserRouter as Router } from 'react-router-dom'
import { buildFakeTrade, TradeSource } from '@komgo/types'
import renderer from 'react-test-renderer'

describe('ReceivableDiscountingTradeView component', () => {
  let defaultProps
  let wrapper: ReactWrapper
  beforeEach(() => {
    defaultProps = {
      company: 'companyId',
      trade: { ...buildFakeTrade({ source: TradeSource.Komgo, sourceId: 'sourceId' }), requiredDocuments: [] },
      open: true,
      index: 'FakePanelIndex',
      changed: false,
      handleClick: jest.fn(),
      tradeMovements: ['movement1', 'movement2'],
      handleEditClicked: jest.fn()
    }
    jest.resetAllMocks()
    wrapper = mount(<ReceivableDiscountingTradeView {...defaultProps} />)
  })

  it('renders correctly', () => {
    expect(renderer.create(<ReceivableDiscountingTradeView {...defaultProps} />).toJSON()).toMatchSnapshot()
  })

  it('should find a MinimalAccordionWrapper', () => {
    const minimalAccordionWrapper = wrapper.find(MinimalAccordionWrapper)
    expect(minimalAccordionWrapper.prop('active')).toEqual(defaultProps.open)
    expect(minimalAccordionWrapper.prop('index')).toEqual(defaultProps.index)
    expect(minimalAccordionWrapper.prop('title')).toEqual('Trade summary')
  })

  it('should find TradeViewData', () => {
    const tradeViewData = wrapper.find(MinimalAccordionWrapper).find(TradeViewData)

    expect(tradeViewData.prop('company')).toEqual(defaultProps.company)
    expect(tradeViewData.prop('hideDropdownIcon')).toEqual(true)
    expect(tradeViewData.prop('trade')).toEqual(defaultProps.trade)
    expect(tradeViewData.prop('tradeMovements')).toEqual(defaultProps.tradeMovements)
  })

  it('should show an edit button if editable', () => {
    expect(
      wrapper
        .find('[data-test-id="edit-trade-request"]')
        .first()
        .exists()
    ).toBeTruthy()
  })

  it('should call handleEditClicked when clicked', done => {
    wrapper
      .find('[data-test-id="edit-trade-request"]')
      .first()
      .simulate('click')

    setTimeout(() => {
      expect(defaultProps.handleEditClicked).toHaveBeenCalled()
      done()
    }, 0)
  })

  it('should not show an edit button if not editable', () => {
    defaultProps.handleEditClicked = undefined
    const testSpecificMount = mount(
      <Router>
        <ReceivableDiscountingTradeView {...defaultProps} />
      </Router>
    )

    expect(
      testSpecificMount
        .find('[data-test-id="edit-trade-request"]')
        .first()
        .exists()
    ).toBeFalsy()
  })
})
