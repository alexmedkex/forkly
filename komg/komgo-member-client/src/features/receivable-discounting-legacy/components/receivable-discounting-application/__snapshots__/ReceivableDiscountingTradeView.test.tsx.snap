// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReceivableDiscountingTradeView component renders correctly 1`] = `
.c0.c0.c0 .komgo-accordion-header {
  height: 53px;
  padding: 15px;
  background-color: #ffffff;
  border: 1px solid #c0cfde;
  border-radius: 4px;
  box-shadow: 0 1px 4px 0 rgba(192,207,222,0.5);
  color: #1c2936;
  font-size: 14px;
  font-weight: 600;
  line-height: 21px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0.c0.c0 .accordion-title-wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 40%;
}

.c0.c0.c0 .accordion-title {
  margin-top: 1px;
}

.c0.c0.c0 .accordion-icon {
  color: #62768d;
  margin: -8px 5px 0px 5px;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: all 0.2s linear;
  transition: all 0.2s linear;
}

.c0.c0.c0 .accordion-status-wrapper {
  margin-top: 1px;
}

.c0.c0.c0 .accordion-button-wrapper {
  margin-top: 1px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  min-width: 20%;
}

.c0.c0.c0 .disabled {
  pointer-events: none;
}

.c0.c0.c0 .lively-divider {
  display: none;
  position: relative;
  top: 14px;
  border-bottom: 1px solid #dbe5ec;
  text-align: center;
  left: 50%;
  width: 0%;
  -webkit-animation: open-line 0.4s forwards;
  animation: open-line 0.4s forwards;
}

.c0.c0.c0 .active .komgo-accordion-header {
  border-radius: 4px 4px 0px 0px;
  border-bottom: 0;
}

.c0.c0.c0 .active .accordion-title {
  margin-top: 0px;
}

.c0.c0.c0 .active .accordion-status-wrapper {
  margin-top: 0px;
}

.c0.c0.c0 .active .accordion-button-wrapper {
  margin-top: 0px;
}

.c0.c0.c0 .active .komgo-accordion-content {
  border-radius: 0px 0px 4px 4px;
  border-top: none;
  border-color: #c0cfde;
  margin: -15px 0px 8px 0px;
}

.c0.c0.c0 .active .accordion-icon {
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
  margin: 0px 5px 0px 5px;
}

.c0.c0.c0 .active .lively-divider {
  display: block;
  left: 1rem;
  width: calc(100% - 2rem);
  top: -15px;
}

.c1.c1.c1 {
  text-transform: uppercase;
  border: none;
  background-color: #f2f5f8;
}

.c2 {
  list-style: none;
  margin: 0;
  padding: 30px 0px 30px 0px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  margin: 0;
  line-height: 2;
  white-space: nowrap;
  margin-bottom: 5px;
  max-width: 720px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c5.c5.c5.c5.c5 .field .selection.dropdown {
  width: 50% !important;
}

.c5.c5.c5.c5.c5 .field.error {
  margin-bottom: 0;
}

.c5.c5.c5.c5.c5 label.inputLabel {
  width: 200px;
  margin-right: 0;
  display: inline-block;
  text-align: right;
  padding-right: 20px;
}

.c4 {
  display: inline-block;
  font-weight: bold;
  width: 200px;
  text-align: right;
  padding-right: 20px;
}

@media (min-width:68.75em) {
  .c2 {
    padding-left: 40px;
  }
}

<div
  bordercolor="#c0cfde"
  className="accordion ui fluid c0"
>
  <div
    className="active title"
    data-test-id="trade-summary-accordion-title"
    onClick={[Function]}
  >
    <div
      className="ui block header komgo-accordion-header"
    >
      <div
        className="accordion-title-wrapper"
      >
        <i
          aria-hidden="true"
          className="chevron right icon accordion-icon"
        />
        <div
          className="accordion-title"
          style={
            Object {
              "marginLeft": "0px",
            }
          }
        >
          Trade summary
        </div>
      </div>
      <div
        className="accordion-status-wrapper"
      />
      <div
        className="accordion-button-wrapper"
      >
        <button
          className="ui button"
          data-test-id="edit-trade-request"
          onClick={[Function]}
          role="button"
        >
          Edit
        </button>
      </div>
    </div>
  </div>
  <div
    className="content active"
    data-test-id="trade-summary-accordion-content"
  >
    <div
      className="lively-divider"
    />
    <div
      className="ui segment komgo-accordion-content"
    >
      <div
        className="accordion ui fluid"
      >
        <div
          className="active title"
          onClick={[Function]}
          style={
            Object {
              "paddingTop": 0,
              "pointerEvents": "none",
            }
          }
        >
          <div
            className="ui block header c1"
          >
            Trade data
          </div>
        </div>
        <div
          className="content active"
        >
          <ul
            className="c2"
          >
            <li
              className="c3"
              data-test-id="header-text"
            >
              <span
                className="c4"
              >
                Trade source
              </span>
              Komgo
            </li>
            <li
              className="c3"
            >
              <span
                className="c4"
              >
                Buyer
              </span>
              
            </li>
            <li
              className="c3"
            >
              <span
                className="c4"
              >
                Buyer's reference
              </span>
              buyerEtrmId
            </li>
            <li
              className="c3"
            >
              <span
                className="c4"
              >
                Seller
              </span>
              
            </li>
            <li
              className="c3"
            >
              <span
                className="c4"
              >
                Credit requirement
              </span>
              Documentary letter of credit
            </li>
            <li
              className="c3"
            >
              <span
                className="c4"
              >
                Reference date
              </span>
              2017/12/31
            </li>
          </ul>
        </div>
        <div
          className="active title"
          onClick={[Function]}
          style={
            Object {
              "pointerEvents": "none",
            }
          }
        >
          <div
            className="ui block header c1"
          >
            Commodity and price
          </div>
        </div>
        <div
          className="content active"
        >
          <ul
            className="c2"
          >
            <li
              className="c3"
            >
              <span
                className="c4"
              >
                Commodity type
              </span>
              Crude oil
            </li>
            <li
              className="c5 c3"
            >
              <span
                className="c4"
              >
                Invoice quantity
              </span>
              Load
            </li>
            <li
              className="c3"
            >
              <li
                className="c5 c3"
              >
                <span
                  className="c4"
                >
                  Quantity
                </span>
                600,000 BBL
              </li>
            </li>
            <li
              className="c5 c3"
            >
              <span
                className="c4"
              >
                Min tolerance
              </span>
              1.25%
            </li>
            <li
              className="c5 c3"
            >
              <span
                className="c4"
              >
                Max tolerance
              </span>
              1.25%
            </li>
            <li
              className="c3"
            >
              <span
                className="c4"
              >
                Price
                 per
                 
                unit
              </span>
              70.02 EUR/BBL
            </li>
            <li
              className="c5 c3"
            >
              <span
                className="c4"
              >
                Payment option
              </span>
              Deferred
            </li>
            <li
              className="c3"
            >
              <span
                className="c4"
              >
                Payment terms
              </span>
              30 calendar days after BL
            </li>
          </ul>
        </div>
        <div
          className="active title"
          onClick={[Function]}
          style={
            Object {
              "pointerEvents": "none",
            }
          }
        >
          <div
            className="ui block header c1"
          >
            Delivery terms
          </div>
        </div>
        <div
          className="content active"
        >
          <ul
            className="c2"
          >
            <li
              className="c3"
            >
              <li
                className="c5 c3"
              >
                <span
                  className="c4"
                >
                  Delivery from
                </span>
                2019/04/03
              </li>
            </li>
            <li
              className="c3"
            >
              <li
                className="c5 c3"
              >
                <span
                  className="c4"
                >
                  Delivery to
                </span>
                2019/04/10
              </li>
            </li>
            <li
              className="c5 c3"
            >
              <span
                className="c4"
              >
                Delivery term (incoterm)
              </span>
              FOB
            </li>
            <li
              className="c5 c3"
            >
              <span
                className="c4"
              >
                Delivery location
              </span>
              Weymouth
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
